src/server/server.cpp

```diff
@@ -1625,7 +1632,29 @@ void server::handle_player_in_game(socket_ptr socket, std::shared_ptr<simple_wml
 
 		if(user_handler_) {
 			const simple_wml::node& m = *g.level().root().child("multiplayer");
-			user_handler_->db_insert_game_info(uuid_, g.db_id(), game_config::wesnoth_version.str(), g.name(), m["mp_scenario"].to_string(), m["mp_era"].to_string(), g.is_reload(), m["observer"].to_bool(), !m["private_replay"].to_bool(), g.has_password());
+
+			// [addon] info handling
+			std::string scenario_addon_id = "";
+			std::string scenario_addon_version = "";
+			std::string era_addon_id = "";
+			std::string era_addon_version = "";
+			for(const auto& addon : m.children("addon")) {
+				for(const auto& content : addon->children("content")) {
+					if(content->attr("type").to_string() == "scenario") {
```

Comment:
Would be more readable to assign `content->attr("type").to_string()` to a variable to access the type that way in each if statement. (Also more efficient if the compiler does not do this internally anyway. Though efficiency is probably no concern here.)

Reply:
Iris mentioned on Discord recently that wesnothd actually is compiled with `-O0`, so making sure the plain C++ code is efficient is definitely a good thing to do since the compiler won't be optimizing anything.

Reply:
> Also more efficient if the compiler does not do this internally anyway

The compiler is usually bad at this type of optimisation since it hard to check whether to_string() might have side effects or not. So it's better to do it by hand..

But i wonder why you call `to_string()` here in the first place instead of comparing attr("type") directly?

Reply:
Doing just `content->attr("type")` returns a string_span, rather than a string.

Reply:
> Doing just content->attr("type") returns a string_span, rather than a string.

Yes, and string_span can be compared directly to "scenario"

Reply:
Updated.